### PR TITLE
Use display name throughout the app

### LINKED ISSUE #246: Use Twitch display names
Right now we just use logins which are all lowercase. We should use the capitalization that the user's Twitch account has. We also need to sync these once in a while with Twitch.

### PR DESCRIPTION
**Pull request by BatedUrGonnaDie**
Closes #246

### CODE REVIEW

File: db/migrate/20170209005256_add_display_name_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddDisplayNameToUsers < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :display_name, :string
```
Comment (glacials) [inline db/migrate/20170209005256_add_display_name_to_users.rb]: Can we call this `twitch_display_name`? I regret calling `name` `name`, as adding other sign in methods later just makes that confusing.

**Comment by BatedUrGonnaDie** [comment]
Looks like to_s was already overridden, so adding display_name onto the front should be ok, yes?

**Comment by glacials** [comment]
Yeah, for sure. Also, can you make sure to test with Chinese, Japanese, or
Korean characters? These are allowed in Twitch display names, so just want
to make sure PostgreSQL handles them fine.

**Comment by BatedUrGonnaDie** [comment]
chrome_2017-02-08_22-40-56

Works ok with everyone's favorite toobou

**Pull request merged**